javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -57,6 +66,106 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
         methodBodyWriter = javaBlock -> javaBlock.line(methodInvocation.toString());
     }
 
+    /**
+     * Parse example node from given parameter, taking into account parameter grouping.
+     *
+     * @param convenienceMethod the convenience method to generate example for
+     * @param proxyMethodExample the proxy method example
+     * @param methodParameter mapped convenience method parameter to protocol(proxy) method parameter
+     * @return example node
+     */
+    private ExampleNode parseNodeFromParameter(ClientMethod convenienceMethod, ProxyMethodExample proxyMethodExample, MethodParameter methodParameter) {
+        if (isGroupingParameter(convenienceMethod, methodParameter)) {
+            // grouping, possible with flattening first
+
+            // group example values into a map
+            Map<String, Object> exampleValue = new HashMap<>();
+            for (MethodTransformationDetail detail : convenienceMethod.getMethodTransformationDetails()) {
+                for (ParameterMapping parameterMapping : detail.getParameterMappings()) {
+                    String serializedParameterName = parameterMapping.getInputParameterProperty().getSerializedName();
+                    ClientMethodParameter parameter = detail.getOutParameter();
+                    if (parameterMapping.getOutputParameterPropertyName() != null) {
+                        // this is a flattened property, so put flattening(real parameter) value
+                        serializedParameterName = detail.getOutParameter().getName();
+                        Map<String, Object> flattenedParameterValue = (Map<String, Object>) ModelExampleUtil.getParameterExampleValue(proxyMethodExample, serializedParameterName, parameter.getRequestParameterLocation());
+                        if (flattenedParameterValue != null) {
+                            exampleValue.putAll(flattenedParameterValue);
+                        }
+                    } else {
+                        exampleValue.put(serializedParameterName,
+                                ModelExampleUtil.getParameterExampleValue(
+                                        proxyMethodExample, serializedParameterName, parameter.getRequestParameterLocation()));
+                    }
```

Comment:
I assume the correctness of the logic depends on that "serialized name" of real parameter would be same as "serialized name" of the group property. This equal is is implicit (as "serialized name" of the group property won't have effect on wire, so strictly speaking it doesn't have to be that value).

If so, recommend to note this in comment.

Reply:
Exactly. The group property's name is actually from real parameter's `serializedName` in emitter. Will add comment about this.